### DISCUSSION TITLE
STTextView cursor and line number bug

### DISCUSSION THREAD

**Discussion by Charliego3** [krzyzanowskim/STTextView#45] [Questions]
As title says.

There is a some bugs:
1. No cursor displayed at all
2. When pressing Enter, the line number appears on the next line and is not the current line.
3. When typing, the content of the last line flashes continuously.
4. When clicking outside the entered text, the cursor does not jump to the last line.

https://github.com/krzyzanowskim/STTextView/assets/26761592/85cf220c-aea4-489d-93b0-e02230dc8295

The code in the example is used [ViewController](https://github.com/krzyzanowskim/STTextView/blob/main/TextEdit/ViewController.swift)

**Comment by krzyzanowskim** [comment]
hm. is there an xcode project to reproduce on my side? do you see missing cursor in demo app?

**Reply by Charliego3** [reply]
Yes, the cursor cannot be seen.
Here is a sample project: [Example](https://github.com/Charliego3/STTextViewExample.git)

**Reply by krzyzanowskim** [reply]
https://github.com/krzyzanowskim/STTextView/assets/758033/fa169c7f-d68e-4b97-898a-497d02180846

it seems to "work on my machine". What is the macOS/Xcode version you use and run it at?

**Reply by Charliego3** [reply]
When running on my `iMac (2017) macOS: Ventura13.6.3, Xcode Version 15.1 (15C65)` and `Macbook Pro (2018) macOS: Ventura13.6.3,  Xcode Version 15.2 (15C500b)`, the screen cannot be displayed normally. I am not sure if this is a system problem. The first screen recording was on the iMac. Above, below is the screen recording on MacBook Pro

https://github.com/krzyzanowskim/STTextView/assets/26761592/c87d49ee-cbd5-44f5-a49d-1d96c466b3ff

**Reply by krzyzanowskim** [reply]
I see. The cursor is different on Sonoma (14), so that's one. I can test that.
The line thing is harder to debug without running on Ventura, they've break and fixed various things between releases.
Thanks for reporting that, I'll try to reproduce it further

**Reply by Charliego3** [reply]
Yes, it's system related, I ran it today on Sonoma 14.3 Xcode 15.2 (15C500b) and everything worked fine, no issues

**Reply by krzyzanowskim** [reply]
that's annoying. I have support for earlier version of insertion point, but again it works on macOS 14, and to test it on macOS 13 I need to install and setup separate environment

**Comment by krzyzanowskim** [comment]
Thank you. I found the issue and fixed it. A fix is on the main branch and will be available with the next release.
